synapse/push/mailer.py

```diff
@@ -197,11 +197,23 @@ def _fetch_room_state(room_id):
         logger.info("Sending email push notification to %s" % email_address)
         # logger.debug(html_text)
 
-        yield sendmail(
-            self.hs.config.email_smtp_host,
-            raw_from, raw_to, multipart_msg.as_string(),
-            port=self.hs.config.email_smtp_port
-        )
+        if self.hs.config.email_smtp_user is not None:
+            yield sendmail(
+                self.hs.config.email_smtp_host,
+                raw_from, raw_to, multipart_msg.as_string(),
+                port=self.hs.config.email_smtp_port,
+                requireAuthentication=True,
+                username=self.hs.config.email_smtp_user,
+                password=self.hs.config.email_smtp_pass,
+                requireTransportSecurity=self.hs.config.require_transport_security
+            )
+        else:
+            yield sendmail(
+                self.hs.config.email_smtp_host,
+                raw_from, raw_to, multipart_msg.as_string(),
+                port=self.hs.config.email_smtp_port,
+                requireTransportSecurity=self.hs.config.require_transport_security
+            )
```

Comment:
It would probably be better to just have one place that calls sendmail: the default for the other params is None, so it should be fine to just pass them through, and presumably requireAuthentication could just be `self.hs.config.email_smtp_user is not None`.